remove bzip2 spkg

dimpase opened the pull request:
a good enough bzip2/libbz2 is readily available on all the systems we support, including "bare" macOS.
Thus it's obsolete, and ripe for removal.

See e.g. [sage-devel](https://groups.google.com/g/sage-devel/c/-ASHfAXqVYo/m/anPQmHxECwAJ) for the relevant context.

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example, -->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->

dimpase committed:
remove bzip2 spkg

tobiasdiez reviewed (approved):
In my opinion, it's a good idea to mark more and more packages as 'prerequisite'. Out of the 400 sage packages, 2/3 thirds are [not up-to-date](https://repology.org/repository/sagemath_develop) - which shows that the maintenance load of the current approach is too heavy (or nobody cares about some packages as they are never installed anyway).

Looking at https://repology.org/project/bzip2/versions, only three linux systems have the older version 1.0.5 (and those are oldtimers like Centos 6 with an eol of 2020). So it should be fine to drop it without a deprecation period.

It might be easier to first decide on a list of systems for which sage should work with only the prerequisite packages installed. Based on this list, it should be straight-forward to decide whether to mark a given sage package as prerequisite or not.

dimpase committed:
add missing ``'s in README.md

dimpase committed:
remove leftover from #39943

inst_patch no longer even defined, so we should remove it from the deps of the base target

dimpase committed:
do libbz2/bzip2 checks in configure

dimpase committed:
more details on bzip2/libbz2 prereq
